src/data_retrieval/cds/common.py

```diff
@@ -25,24 +28,33 @@ def get_dates(year: int) -> str:
     return dates
 
 
-def download_cds(name: str, metadata: Dict[str, Any], file_path: str):
+def download_cds(name: str, metadata: Dict[str, Any]) -> BytesIO:
```

Comment:
Do not kill the local download functionality, but add another argument to distinguish between local and stream download.
You can return `None` for local file download so the type hinting would look like this
```python
def download_cds(name: str, metadata: Dict[str, Any], file_path: Optional[str] = None) -> Optional[BytesIO]:
    if file_path:
         # download locally
         return None
    else:
        # get stream
        return BytesIO
```